### PR TITLE
Fix Wextra-semi warnings

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @EikanWang @voznesenskym @penguinwu @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec @xmfan @yf225

### CODE REVIEW

File: c10/core/SymInt.h [hunk 1]
```diff
@@ -93,7 +93,7 @@ class C10_API SymInt {
     // https://stackoverflow.com/questions/42534749/signed-extension-from-24-bit-to-32-bit-in-c
     uint64_t extended_bits = (unextended_bits ^ sign_bit_mask) - sign_bit_mask;
     return static_cast<SymNodeImpl*>(
-        // NOLINTNEXTLINE(performance-no-int-to-ptr)
+        // NOLINTNEXTLINE
```
Comment (Skylion007) [inline c10/core/SymInt.h:96]: Why? Specific nolintnextlines are good (and you can specify multiple of them if needed).

Comment (cyyever) [reply c10/core/SymInt.h:96]: There is another one to suppress that makes the line to long to format...

Comment (Skylion007) [reply c10/core/SymInt.h:96]: Can you glob them together at least?

Comment (cyyever) [reply c10/core/SymInt.h]: Fixed.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**